src/ecmult_gen_compute_table_impl.h

```diff
@@ -10,74 +10,96 @@
 #include "ecmult_gen_compute_table.h"
 #include "group_impl.h"
 #include "field_impl.h"
+#include "scalar_impl.h"
 #include "ecmult_gen.h"
 #include "util.h"
 
-static void secp256k1_ecmult_gen_compute_table(secp256k1_ge_storage* table, const secp256k1_ge* gen, int bits) {
-    int g = ECMULT_GEN_PREC_G(bits);
-    int n = ECMULT_GEN_PREC_N(bits);
+static void secp256k1_ecmult_gen_compute_table(secp256k1_ge_storage* table, const secp256k1_ge* gen, int blocks, int teeth, int spacing) {
+    size_t points = ((size_t)1) << (teeth - 1);
+    size_t points_total = points * blocks;
+    secp256k1_ge* prec = checked_malloc(&default_error_callback, points_total * sizeof(*prec));
+    secp256k1_gej* ds = checked_malloc(&default_error_callback, teeth * sizeof(*ds));
+    secp256k1_gej* vs = checked_malloc(&default_error_callback, points_total * sizeof(*vs));
+    secp256k1_gej u;
+    size_t vs_pos = 0;
+    secp256k1_scalar half = secp256k1_scalar_zero;
+    int block, i;
 
-    secp256k1_ge* prec = checked_malloc(&default_error_callback, n * g * sizeof(*prec));
-    secp256k1_gej gj;
-    secp256k1_gej nums_gej;
-    int i, j;
-
-    VERIFY_CHECK(g > 0);
-    VERIFY_CHECK(n > 0);
-
-    /* get the generator */
-    secp256k1_gej_set_ge(&gj, gen);
-
-    /* Construct a group element with no known corresponding scalar (nothing up my sleeve). */
+    /* u is the running power of two times gen we're working with, initially gen/2. */
```

Comment:
Add `VERIFY_CHECK(points_total > 0)` here to tell GCC that the loop is at least executed once.

See https://github.com/real-or-random/secp256k1/commit/854723f1e30d0e7e71967d0c225645275d169883#diff-2d5b5da541c539e4be4fc1506b554895bd68ef20bf8ae17879c8af6c948dbe54 for working CI runs on MacOS GCC.

Reply:
Yep, and I had to find this [GCC bug](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=100417) to understand that GCC talks about the *memory pointed to by* `ret` (inside `checked_malloc`) being uninitialized, and not `ret` itself...